File: EIDSS v5/android.java/workspace/EIDSS/src/com/bv/eidss/LocalLoginActivity.java
package com.bv.eidss;

import com.bv.eidss.data.EidssDatabase;
import android.app.Activity;
import android.app.Dialog;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class LocalLoginActivity extends Activity {
    private final int PASSWORD_INVALID_DIALOG_ID = 1;

	private Activity _this;
	public LocalLoginActivity() {
    	_this = this;
	}
    
	@Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTitle(R.string.TitleLocalLogin);
        setContentView(R.layout.local_login_layout);

    	findViewById(R.id.OkButton).setOnClickListener(new View.OnClickListener() {
			@SuppressWarnings("deprecation")
			public void onClick(View v) {
	            EditText pwd = (EditText)_this.findViewById(R.id.LoginPassword);
                EidssDatabase db = new EidssDatabase(_this);
                String password = db.Options().getLocalPassword();
                db.close();
                if (pwd.getText().toString().compareTo(password) != 0)
                {
                	showDialog(PASSWORD_INVALID_DIALOG_ID);
                	return;
                }

                _this.setResult(RESULT_OK);
				_this.finish();
			}
		});
    	findViewById(R.id.CancelButton).setOnClickListener(new View.OnClickListener() {
			public void onClick(View v) {
				_this.setResult(RESULT_CANCELED);
				_this.finish();
			}
		});
    }

	@Override
    protected Dialog onCreateDialog(int id) {
        switch (id)
        {
	        case PASSWORD_INVALID_DIALOG_ID:
                return EidssAndroidHelpers.ErrorDialog(this, getResources().getString(R.string.PasswordInvalid));
        }
        return null;
    }
	
}

File: EIDSS v5/android/EIDSS/LocalLoginActivity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using eidssdroid.model.Database;

namespace EIDSS
{
    [Activity]
    public class LocalLoginActivity : Activity
    {
        private const int PASSWORD_INVALID_DIALOG_ID = 1;

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            SetTitle(Resource.String.TitleLocalLogin);
            SetContentView(Resource.Layout.LocalLoginLayout);
            Button buttonOk = FindViewById<Button>(Resource.Id.OkButton);
            Button buttonCancel = FindViewById<Button>(Resource.Id.CancelButton);
            EditText pwd = FindViewById<EditText>(Resource.Id.LoginPassword);

            buttonOk.Click += delegate
            {
                using (EidssDatabase db = new EidssDatabase(this))
                {
                    string password = db.Options().strLocalPassword;
                    if (pwd.Text != password)
                    {
                        ShowDialog(PASSWORD_INVALID_DIALOG_ID);
                    }
                    else
                    {
                        SetResult(Result.Ok);
                        this.Finish();
                    }
                }
            };

            buttonCancel.Click += delegate
            {
                SetResult(Result.Canceled);
                this.Finish();
            };
        }

        protected override Dialog OnCreateDialog(int id)
        {
            switch (id)
            {
                case PASSWORD_INVALID_DIALOG_ID:
                    return EidssAndroidHelpers.ErrorDialog(this, Resources.GetString(Resource.String.PasswordInvalid));
            }
            return null;
        }

    }
}